src/coord/src/catalog.rs

```diff
@@ -515,17 +516,48 @@ impl CatalogState {
         replica_id: ReplicaId,
         config: ConcreteComputeInstanceReplicaConfig,
     ) {
+        let replica = ComputeInstanceReplica {
+            config,
+            process_status: HashMap::new(),
+        };
         let compute_instance = self.compute_instances_by_id.get_mut(&on_instance).unwrap();
         assert!(compute_instance
             .replica_id_by_name
             .insert(replica_name, replica_id)
             .is_none());
         assert!(compute_instance
             .replicas_by_id
-            .insert(replica_id, config)
+            .insert(replica_id, replica)
             .is_none());
     }
 
+    fn try_insert_compute_instance_status(
```

Comment:
Yes, it's possible that the orchestrator reports status changes for replicas that we have already dropped in the catalog, since we are not waiting for the replica to be completely gone before removing our bookkeeping of it.

I think I prefer adding helpful comments here over adding more checking to the propagation of `Action::UpdateComputeInstanceStatus`. The latter would lead to more complicated code in the coordinator in exchange for less complicated code here, but it seems to me that reducing complexity in the coordinator is more important. The argument would be different if propagating `Action::UpdateComputeInstanceStatus` had a noticeable performance impact, but I don't think that's the case.